### ISSUE TITLE
False-positive arity error: `fn` parameter shadowing a `let` binding is ignored by the arity checker

### ISSUE THREAD

**Issue by se-neaxplore** [clj-kondo/clj-kondo#2952]
## Version

```
clj-kondo v2026.05.25
```

## Problem

When a `let` binding is shadowed by a `fn` parameter of the same name, clj-kondo's
arity checker resolves calls to the shadowed local against the **outer** `let` binding
instead of the shadowing parameter. This produces a false-positive arity error.

## Repro

```clojure
(let [x (fn [_] _)
      y (fn [x] (x))]
  (y #(prn "No argument")))
```

Lint it:

```
$ clj-kondo --lint bug.clj
fringe_linter_bug.clj:2:7:  warning: unused binding x
fringe_linter_bug.clj:3:17: error:   x is called with 0 args but expects 1
linting took 14ms, errors: 1, warnings: 1
```

## Expected

No arity error.

Inside `(fn [x] (x))`, the `x` in `(x)` refers to that `fn`'s **own parameter**, which
shadows the outer `let` binding `x`. The parameter has unknown arity, so `(x)` cannot be flagged. At runtime the form evaluates without error:

## Actual

clj-kondo binds the `x` in `(x)` to the outer `let` value `(fn [_] _)` (arity 1) and
reports `x is called with 0 args but expects 1`.

**Comment by borkdude** [comment]
You closed this. Does it mean it's fixed?

**Comment by se-neaxplore** [comment]
Correct, I wasn't using the latest version, my bad. And you had already fixed it here https://github.com/clj-kondo/clj-kondo/issues/2854
